### PR TITLE
[ned14-internal-quickcpplib] Update port @2022-07-04

### PR DESCRIPTION
**Pull request by BurningEnlightenment**
**Describe the pull request**

- #### What does your PR fix?
  Updates quickcpplib, integrates the upstream dependency switch from `gsl-lite` to `span-lite`.

  Additionally I applied the feedback wrt polyfill features from #25115 

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  Set the Windows arm64 baseline to fail, because it is caused by an upstream bug https://github.com/ned14/quickcpplib/issues/32

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  Yes

### CODE REVIEW

**Comment by BurningEnlightenment** [comment]
I can't reproduce the linux build failure locally and I don't have access to an windows arm64 device. So I need access to the (cmake) build logs.

**Comment by dg0yt** [comment]
> I can't reproduce the linux build failure locally and I don't have access to an windows arm64 device. So I need access to the (cmake) build logs.

You can click through from Checks to Azure pipelines to [artifacts](https://dev.azure.com/vcpkg/public/_build/results?buildId=74594&view=artifacts&pathAsName=false&type=publishedArtifacts), and download the failure logs. The menu for download appears behind the three dots on the right side of an artifact, when the mouse is over that line.

**Comment by BurningEnlightenment** [comment]
Ah, I missed the artifacts link. 

Ok, so the gcc version on linux is too old for `std::span` and would need the `polyfill-cxx20` feature. Is there a way to enable this just for the CI?

I will set the Windows arm64 baseline to fail, because it is caused by an upstream bug ned14/quickcpplib#32

**Comment by dg0yt** [comment]
> Ok, so the gcc version on linux is too old for `std::span` and would need the `polyfill-cxx20` feature. Is there a way to enable this just for the CI?

As an idea, for feedback from the core maintainers: A CI-only activation could be achieved via a test port (e.g. a new empty `vcpkg-ci-polyfill` port) in `scripts/test-ports` which depends on `quickcpplib[polyfill-cxx20]` for linux.

File: scripts/ci.baseline.txt [hunk 1]
```diff
@@ -563,6 +563,8 @@ linenoise-ng:x64-uwp=fail
 live555:arm-uwp=fail
 live555:x64-osx=fail
 live555:x64-uwp=fail
+llfio:arm64-windows=fail
+llfio:x64-linux=fail
```
Comment (Adela0814) [inline scripts/ci.baseline.txt:567]: Please add a comment here explaining why this is added.

**Review by Adela0814 — CHANGES_REQUESTED** [review]

**Comment by BurningEnlightenment** [comment]
@Adela0814 is this sufficient?

**Review by Adela0814 — APPROVED** [review]

**Pull request merged**